drivers: dma: silabs: Implementation of chan_filter and chan_release API with DMADRV functions.

Martinhoff-maker opened the pull request:
The goal of this patch is to maintain synchronization on DMA channel allocation between HAL DMA driver and Zephyr Silabs DMA driver.

teburd commented on drivers/dma/dma_silabs_ldma.c:
```diff
@@ -502,7 +513,9 @@ static DEVICE_API(dma, dma_funcs) = {
 	.start = dma_silabs_start,
 	.stop = dma_silabs_stop,
-	.get_status = dma_silabs_get_status
+	.get_status = dma_silabs_get_status,
```
Please review and ensure the requirements have been met for these APIs, I have no insight into the HAL implementation of these functions so I can't definitively say if the requirements have been met.

Martinhoff-maker committed:
west.yml: update hal_silabs to latest version

Update hal_silabs to include emdrv/dmadrv.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

Martinhoff-maker committed:
drivers: dma: silabs: implementation of chan_filter and chan_release

Implementation of chan_filter and chan_release allows to maintain
synchronization about DMA channel allocation between hal DMA driver and
zephyr DMA driver.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

Martinhoff-maker committed:
modules: hal_silabs: Update CMake with new EMDRV hal

Update Cmake to include hal EMDRV header
It also adds dmadrv source when CONFIG_SOC_GECKO_LDMA is selected

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>
